src/os/host-start.r

```diff
@@ -678,3 +628,66 @@ comment [
     return :host-repl
 ]
 
+
+; Define REPL! object for skinning - stub for elsewhere?
+;
+
+repl!: make object! [
+    repl: true
+    counter: 0
+    last-result: _ ;-- last evaluated result (sent by HOST-REPL)
+
+    ; Called on every line of input by HOST-REPL in %os/host-repl.r
+    ;
+    cycle: does [
+        if zero? ++ counter [print-greeting] ;-- only load skin on first cycle
+        counter
+    ]
+
+    ;; APPEARANCE (can be overridden)
+
+    prompt:   {>> }
+    result:   {== }
+    warning:  {!! }
+    error:    {** }                ;; not used yet
+    info:     to-string #{e29398}  ;; info sign!
+    greeting: _
+    print-prompt:   proc []  [print/only prompt]
+    print-result:   proc []  [print unspaced [result last-result]]
+    print-warning:  proc [s] [print unspaced [warning reduce s]]
+    print-error:    proc [e] [print e]
+    print-info:     proc [s] [print [info space space reduce s]]
+    print-greeting: proc []  [boot-print greeting]
+    print-gap:      proc []  [print-newline]
+
+    ;; BEHAVIOR (can be overridden)
+    
+    input-hook: func [
+        {Receives line input, parse/transform, send back to repl eval}
+        s
+    ][
+        s
+    ]
+
+    dialect-hook: func [
+        {Receives code block, parse/transform, send back to repl eval}
+        s
+    ][
+        s
```

Comment:
Thanks for the feedback... at least you know it's there now :-)

It's a little "safer", but I don't think of that safety from a "security" standpoint for the reasons I've mentioned.  Rebol's security model isn't going to come from this granularity.  You just don't get a lot of security in a language which does not distinguish functions from inert values at the callsite (unless your language has other ways of preventing a function from "doing anything").  "The most freeform computer language ever invented" isn't going to coincidentally somehow also be "The most secure computer language ever invented" :-)

So it's more about just helping avoid confusion.  But the reason I ask is I'm not sure how much confusion it actually has helped with.  It's a tough balance when it comes to deciding how laissez-faire to be, and I think my leanings are moving more toward saying that I can't think of all that many bugs it has stopped from people whose intentions weren't malicious.

It's been a trial period.  It doesn't hurt that it's there, as it can be relaxed later.  Just something to be aware of and think about.